test/jdk/jdk/crac/ContextOrderTest.java

```diff
@@ -294,12 +283,18 @@ public String toString() {
         private final Context<R> childContext;
         private boolean first = true;
 
-        private CreatingResource(List<String> recorder, Priority priority, String id, Context<R> childContext, Priority childPriority) {
+        private CreatingResource(List<String> recorder, Priority priority, String id, Priority childPriority) {
             super(recorder, priority, id);
-            this.childContext = childContext;
+            this.childContext = (Context<R>) childPriority.getContext(); // XXX
```

Comment:
`// XXX`

Reply:
Indeed, I feel something wrong with the resulting types in the test, but not sure how to fix.